com.unity.formats.fbx/Editor/FbxExporter.cs

```diff
@@ -1023,19 +1023,46 @@ SkinnedMeshRenderer unitySkin
         /// <returns>The bind pose.</returns>
         /// <param name="unityBone">Unity bone.</param>
         /// <param name="bindPoses">Bind poses.</param>
-        /// <param name="boneDict">Dictionary of bone to index.</param>
-        /// <param name="skinnedMesh">Skinned mesh.</param>
+        /// <param name="boneInfo">Contains information about bones and skinned mesh.</param>
         private Matrix4x4 GetBindPose(
             Transform unityBone, Matrix4x4[] bindPoses,
-            Dictionary<Transform, int> boneDict, SkinnedMeshRenderer skinnedMesh
-        ){
+            ref SkinnedMeshBoneInfo boneInfo
+        )
+        {
+            var boneDict = boneInfo.boneDict;
+            var skinnedMesh = boneInfo.skinnedMesh;
+            var boneToBindPose = boneInfo.boneToBindPose;
+
             Matrix4x4 bindPose;
+            if(boneToBindPose.TryGetValue(unityBone, out bindPose))
+            {
+                return bindPose;
+            }
+
             int index;
-            if (boneDict.TryGetValue (unityBone, out index)) {
-                bindPose = bindPoses [index];
-            } else {
-                bindPose = unityBone.worldToLocalMatrix * skinnedMesh.transform.localToWorldMatrix;
+            if (boneDict.TryGetValue(unityBone, out index))
+            {
+                bindPose = bindPoses[index];
+                boneToBindPose.Add(unityBone, bindPose);
+                return bindPose;
             }
+
+            if(unityBone == skinnedMesh.rootBone || unityBone.parent == null)
+            {
+                // there is no bone above this object with a bindpose, calculate bindpose relative to skinned mesh
+                bindPose = (unityBone.worldToLocalMatrix * skinnedMesh.transform.localToWorldMatrix);
+                boneToBindPose.Add(unityBone, bindPose);
+                return bindPose;
+            }
+
+            // get the bindpose of the parent
+            var parentPose = GetBindPose(unityBone.parent, bindPoses, ref boneInfo);
+            // get the local transformation matrix of the bone, then transform it into
+            // the global transformation matrix with the parent in the bind pose.
+            bindPose = parentPose.inverse * Matrix4x4.TRS(unityBone.localPosition, unityBone.localRotation, unityBone.localScale);
+            // bindpose is the inverse of the transformation matrix
+            bindPose = bindPose.inverse;
```

Comment:
We can simplify this and save an inverse (and thus reduce numerical noise): recall the identity `(AB)^{-1} = B^{-1}A^{-1}`

Currently you have
`bindPose = (parentPose.inverse * TRS).inverse`
which should be equal to
`bindPose = TRS.inverse * parentPose.inverse.inverse`
which is mathematically
`bindPose = TRS.inverse * parentPose`
...
...
I think.